tests/unit/test_bids_utils.py

```diff
@@ -81,3 +86,66 @@ def test_bidsmap_scans_run_plus():
     scans = [("1", "run+"), ("2", "run+"), ("3", "run+")]
 
     assert bidsmap_scans(scans) == [("1", "run-01"), ("2", "run-02"), ("3", "run-03")]
+
+
+def test_bidsify_dicom_headers(mocker):
+    """Test bidsify_dicom_headers without ProtocolName field"""
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = False
+
+    dcmread = mocker.patch("pydicom.dcmread")
+    dcmread.return_value = dataset
+
+    bidsify_dicom_headers("filename", "foo")
+
+    assert dataset.data_element.called == False
+
+
+def test_bidsify_dicom_headers_with_protocol_name(mocker):
+    """Test bidsify_dicom_headers with ProtocolName match"""
+    series_description = "foo"
+
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = True
+    dataset.data_element.return_value = mocker.Mock(value=series_description)
+
+    dcmread = mocker.patch("pydicom.dcmread")
+    dcmread.return_value = dataset
+
+    bidsify_dicom_headers("filename", series_description)
+
+    dataset.data_element.assert_called_once_with("ProtocolName")
```

Comment:
5. Here we check to make sure that the `dataset.data_element` has been called only once. We want to make sure it's only called once because we don't expect any additional calls if the `dataset.data_element("ProtocolName").value` already is the same as `series_description`